packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -771,6 +779,7 @@ export class Project extends ProjectBase {
       triggers: sourceConfig.buildTriggers,
       sourceVersion: sourceConfig.sourceVersion,
       vpcConfig: this.configureVpc(props),
+      logsConfig: props.logging && this.renderLogs(props),
```

Comment:
Just call this like this: `this.renderLoggingConfiguration(props.logging)`, and let the method handle the `undefined` case.